conan/tools/files/files.py

```diff
@@ -491,3 +493,20 @@ def collect_libs(conanfile, folder=None):
                     result.append(name)
     result.sort()
     return result
+
+
+def swap_child_folder(parent_folder, child_folder):
```

Comment:
Name to be improved. Suggestions welcome.

Reply:
Maybe a generic `move_folder_contents(dst_folder, folder, clean_first=True)` ?

Reply:
I think maybe we can split the operation in two, one for removing all folders but the one we want and other for moving the folder... or `move_folder_contents(dst_folder, folder, clean_first=True)` maybe also more explicit, I'm ok with that too, maybe `clean_other` instead of `clean_first`?

Reply:
I am annotating it to NOT document, so we can think about file methods in a more broad perspective, this PR is focused on the Git integration for sources, this is a very minor detail, just to prove it is possible to implement the mono-repo approach